cmd/kubeadm/app/phases/addons/proxy/proxy.go

```diff
@@ -70,41 +70,30 @@ func EnsureProxyAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Inte
 		return fmt.Errorf("error when parsing kube-proxy daemonset template: %v", err)
 	}
 
-	if err = createKubeProxyAddon(proxyConfigMapBytes, proxyDaemonSetBytes, client); err != nil {
+	if err := createKubeProxyAddon(proxyConfigMapBytes, proxyDaemonSetBytes, client); err != nil {
 		return err
 	}
-	fmt.Println("[addons] Applied essential addon: kube-proxy")
-
-	k8sVersion, err := version.ParseSemantic(cfg.KubernetesVersion)
-	if err != nil {
-		return fmt.Errorf("couldn't parse kubernetes version %q: %v", cfg.KubernetesVersion, err)
-	}
-	if err = CreateRBACRules(client, k8sVersion); err != nil {
+	if err := CreateRBACRules(client); err != nil {
 		return fmt.Errorf("error when creating kube-proxy RBAC rules: %v", err)
 	}
 
+	fmt.Println("[addons] Applied essential addon: kube-proxy")
 	return nil
 }
 
 // CreateServiceAccount creates the necessary serviceaccounts that kubeadm uses/might use, if they don't already exist.
 func CreateServiceAccount(client clientset.Interface) error {
-	sa := v1.ServiceAccount{
+
+	return apiclient.CreateOrUpdateServiceAccount(client, &v1.ServiceAccount{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      kubeadmconstants.KubeProxyServiceAccountName,
+			Name:      KubeProxyServiceAccountName,
 			Namespace: metav1.NamespaceSystem,
 		},
-	}
-
-	if _, err := client.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Create(&sa); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			return err
-		}
-	}
-	return nil
+	})
 }
 
 // CreateRBACRules creates the essential RBAC rules for a minimally set-up cluster
-func CreateRBACRules(client clientset.Interface, k8sVersion *k8sversion.Version) error {
+func CreateRBACRules(client clientset.Interface) error {
```

Comment:
We agreed with @andrewrynhard that it might be a nice pattern to keep for consumers at golang code:
 - EnsureFooAddon
 - CreateRBACRules
 - CreateServiceAccount

where the last two are optional for the phase to provide